docs/adr/2022-04-26__alerting.md

```diff
@@ -0,0 +1,295 @@
+
+It is not always possible to define a specific threshold, instead previously recorded data should be used to detect significant deviations.
+
+This check is the most complicated and computation-intensive one with potentially the highest number of false positives. There are a lot of different anomaly detection algorithms out there, but for OpMon the idea is to detect these anomalies using the exponentially weighted moving average (EWMA):
+
+```math
+EWMA_t = a * v_t + (1 - a) * EWMA_{t-1}
+```
+
+* `a`: configurable weight; larger values result in more recent recorded values having a larger impact on the moving average (= they will have a higher weight)
+* `v`: value of a specific metric
+* `EWMA_0`: the starting point, which is equal to the first value recorded for the metric
+* `t`: observation day
+
+EWMA considers all historic values but puts a lot of weight on more recent values, older values are given less weight. This should ensure that spikes recorded in the past are less likely to trigger future alerts.
+
+Control limits are determined by calculating the standard deviation:
+
+```math
+stdev(v)_t = \sqrt(EWMA(v^2)_t - EWMA(v)^2_t)
+```
+
+If the recorded values for a metric exceeds `EWMA_t + stdev(v)_t` or is below `EWMA_t - stdev(v)_t`, then a change is present.
+
+This method is memory and CPU efficient and can be easily implemented in SQL. It only works for metrics with no seasonality. In the case of OpMon most operational metrics, like memory consumption or different load times, are not influenced by different week days or holidays.
+
+A few drawbacks of this methods are that over time the control limits can get temporarily inflated after a spike and changes could go unnoticed afterwards. Slow changes will go unnoticed, for example if a metric slowly degrades over a long period of time, the average will adjust to it instead of noticing the change.
+Another issue is that for computing the percentiles for each metric, values need to get bucketed. Depending on the resolution of these buckets, value changes do not show up as steady changes but instead as sudden jumps. This could potentially trigger this check if the resolution of the bucketes is low.
+
+This check needs to be configured for metrics like:
+
+```toml
+[alert]
+
+[alert.crash_changes] # new alert
+type = "ema"
+precentiles = [50, 90] # percentiles that should be considered
+weight = 0.4    # weight factor    
+probes = [      # list of metrics to check
+    "startup_crashes", 
+    "main_crashes", 
+    "oom_crashes"
+]
+```
+
+
+## Displaying Changes and Notifications
+
+Significant changes are displayed in a table on the OpMon dashboard. For example:
+
+<img src="./images/alerting_table.png" width="500"/>
+
+Users can opt-in to receiving notifications via email by creating a new alert on the table in Looker:
+
+<img src="./images/alerting_notification.png" width="500"/>
+
+
+
+## Alternatives Considered
+
+### More Advanced Time Series Modelling
```

Comment:
My vote is to start simple. In time series analysis, often a simple moving average approach can get 80-90% of the accuracy that a much more complicated approach can achieve.